en/companion_computer/pixhawk_rpi.md

```diff
@@ -0,0 +1,259 @@
+# Using the Raspberry Pi with the Pixhawk Controllers
+
+This section describes how to connect the Pixhawk to the Raspberry Pi ("RPi"). Both the software setup and the wire connections between the hardware are presented. 
+
+There are different ways to connect the RPi with the Pixhawk:
+
+1. Via Ethernet port.
+
+2. Serial connection: via `TELEM` port or via `USB` port.
+
+The connection via `TELEM` port can be done with a FTDI Chip USB-to-serial adapter board or directly at the RPi's TX/RX pins. The following steps describes the connections using the TX/RX pins, since no additional board is required.
+
+There are different models of Pixhawk and RPi, therefore, the main concepts are covered and illustrated in order to give the user the necessary knowledge for any other configuration. The first step is the identification of the `TELEM 2` and the `USB` ports in the Pixhawk. The image illustrates the ports in the Pixhawk-6C Model.
+
+<img src="../../assets/companion_computer/pixhawk_rpi/pixhawk_6c.png" width="402" height="452" />
+
+## PX4 setup on Pixhawk
+
+Connect the Pixhawk with the laptop/desktop via `USB` port and check, at the QGroundControl, if the firmware (PX4) is updated. 
+The firmware update can be done in the QGroundControl or following the next steps:
+
+Run in the terminal:
+
+```
+sudo apt -y install gcc-arm-none-eabi make stlink-tools
+```
+
+Clone the last PX4 version and run the script setup:
+
+```
+git clone https://github.com/PX4/PX4-Autopilot.git --recursive 
+PX4-Autopilot/Tools/setup/ubuntu.sh
+```
+
+Build it on the Pixhawk. On Linux, the default name of a USB connection is `/dev/ttyACM0`:
```

Comment:
This reads as "Build PX4 on Pixhawk" - you don't mean that right?

Re "On Linux, the default name of a USB connection is `/dev/ttyACM0`:" and then `sudo chmod a+rw /dev/ttyACM0`.

Why are you doing this? I mean on my laptop I just connect the USB port to PX4 and attach QGC and it magically works.
Is this required because the USB port might be doing something else on your laptop?

Reply:
> This reads as "Build PX4 on Pixhawk" - you don't mean that right?
 
Yes, I mean that. 

> Re "On Linux, the default name of a USB connection is `/dev/ttyACM0`:" and then `sudo chmod a+rw /dev/ttyACM0`.
> 
> Why are you doing this? I mean on my laptop I just connect the USB port to PX4 and attach QGC and it magically works. Is this required because the USB port might be doing something else on your laptop?

To build the PX4 on the Pixhawk we need to give permission to write on it .... Maybe in your Ubuntu it is the default, but usually I need to change that permission when working with USB devices in Ubuntu (PS.: the same with Arduino for example).

Reply:
>> This reads as "Build PX4 on Pixhawk" - you don't mean that right?
> Yes, I mean that.

As above, no you don't - you mean "for Pixhawk, on your Ubuntu laptop". Building PX4 in Pixhawk would mean that you had Ubuntu running on the Pixhawk and then you built PX4 using the Pixhawk as the computer. Not possible.

> To build the PX4 on the Pixhawk we need to give permission to write on it .... Maybe in your Ubuntu it is the default, but usually I need to change that permission when working with USB devices in Ubuntu (PS.: the same with Arduino for example).

That's clear. Normally I do nothing special to write - I just install the toolchain and do upload. But this is good. What might be good is to try the upload step without this and see what the output is. Then we could say to users "if you get this error, then you need to do this to get the port to work?"

Reply:
This is clear thank you.

Reply:
However I think for the instructions we should update to just point to the existing setup instructions. IF there is a problem that you need to use the command `sudo chmod a+rw /dev/ttyACM0` then we need to update our general Ubuntu documentation to explain this and link to that. Otherwise other people will be stuck who don't read your instructions. Hope that makes sense.